chapter07/axp/ex01

```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/env perl
+
+use 5.014;
+use warnings;
+use File::Find;
+use Time::ParseDate;
+
+my $monday = parsedate( "last Monday at 00:00:00" );
+my $tuesday = $monday + ( 24 * 60 * 60 );
+my ( $gather, $yield ) = gather_mtime_between( $monday, $tuesday );
+
+find( $gather, @ARGV ? @ARGV : ('.') );
+say localtime( (stat)[9] ) . ": $_" foreach ( $yield->() );
+
+sub gather_mtime_between {
+	my ( $start, $end ) = @_;
+	my @files;
```

Comment:
Indeed, every time `gather_mtime_between` is invoked, a new list is created. This is consistent with what we want, as we invoke this method only once to get references to the `$gather` and `$yield` subs, then call these subs as often as necessary.

If we were to call `gather_mtime_between` again, we'd get a separate `@files` list, initialised empty, and two new versions of the subs tied to the new list rather than the old one.